source/tests/system/nidmm_driver_api_tests.cpp

```diff
@@ -0,0 +1,282 @@
+#include <gtest/gtest.h>
+
+#include "nidmm/nidmm_library.h"
+#include "nidmm/nidmm_service.h"
+
+namespace ni {
+namespace tests {
+namespace system {
+
+namespace dmm = nidmm_grpc;
+
+const int kDMMDriverApiSuccess = 0;
+
+class NiDMMDriverApiTest : public ::testing::Test {
+    protected:
+        NiDMMDriverApiTest()
+        {
+            ::grpc::ServerBuilder builder;
+            session_repository_ = std::make_unique<nidevice_grpc::SessionRepository>();
+            nidmm_library_ = std::make_unique<dmm::NiDMMLibrary>();
+            nidmm_service_ = std::make_unique<dmm::NiDMMService>(nidmm_library_.get(), session_repository_.get());
+            builder.RegisterService(nidmm_service_.get());
+
+            server_ = builder.BuildAndStart();
+            ResetStub();
+        }
+
+        virtual ~NiDMMDriverApiTest() {}
+
+        void SetUp() override
+        {
+            initialize_driver_session();
+        }
+
+        void TearDown() override
+        {
+            close_driver_session();
+        }
+
+        void ResetStub()
+        {
+            channel_ = server_->InProcessChannel(::grpc::ChannelArguments());
+            nidmm_stub_ = dmm::NiDMM::NewStub(channel_);
+        }
+
+        std::unique_ptr<dmm::NiDMM::Stub>& GetStub()
+        {
+            return nidmm_stub_;
+        }
+
+        int GetSessionId()
+        {
+            return driver_session_->id();
+        }
+
+        void expect_api_success(int error_status)
+        {
+            EXPECT_EQ(kDMMDriverApiSuccess, error_status) << get_error_message(error_status);
```

Comment:
I would recommend not combining a server call with an assert here, especially since that's not the server code being tested. Create a separate test for `get_error_message`.